ext/soap/soap.c

```diff
@@ -3209,10 +3209,9 @@ PHP_METHOD(SoapClient, __setSoapHeaders)
 	} else if (Z_TYPE_P(headers) == IS_ARRAY) {
 		zval *default_headers;
 
+		zend_hash_str_del(Z_OBJPROP_P(this_ptr), "__default_headers", sizeof("__default_headers")-1);
```

Comment:
Thanks, I assumed we needed to remove the property before attempting to add one with the same name. I'll remove this line now